akka-testkit-typed/src/main/scala/akka/testkit/typed/TestKit.scala

```diff
@@ -17,6 +17,14 @@ import scala.util.control.NoStackTrace
  */
 case class TE(message: String) extends RuntimeException(message) with NoStackTrace
 
+/**
+ * Not for user extension.
+ *
+ * Instances are available from `FishingOutcomes` in the respective dsls: [[akka.testkit.typed.scaladsl.FishingOutcomes]]
```

Comment:
:+1:  to separate APIs. That is what we said we'd do from now on wasn't it? Separate interfaces with one impl. Would be nice to remove the finite durations from the java api as well.

Reply:
Created #24542 to track separating the APIs